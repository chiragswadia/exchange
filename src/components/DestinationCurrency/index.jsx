import React from 'react';
import PropTypes from 'prop-types';
import cns from 'classnames';
import CurrencySelector from '../CurrencySelector';

import cn from './styles.module.css';

function DestinationCurrency(props){
    const onCurrencySelect = destinationCurrency => {
        const { onChange } = props;

        if (onChange) {
            onChange(destinationCurrency);
        }
    };

    const renderAmount = () => {
        const { amount, rate } = props;
    
        if (!rate || !amount) {
          return (
            <div className={cn.value}/>
          );
        }
    
        const value = (amount * rate).toFixed(2);
    
        return (
          <div className={cns(cn.value, {[cn.valueLong]: value.length > 7})}>
            { value }
          </div>
        );
    }

    const { destinationCurrency, wallet } = props;
    const currencies = Object.keys(wallet);

    return (
        <div className={cn.root}>
            <CurrencySelector
                wallet={wallet}
                selectedCurrency={destinationCurrency}
                currencies={currencies}
                onChange={onCurrencySelect}
            />
            <div className={cn.amount}>
                { renderAmount() }
            </div>
        </div>
    );
}

DestinationCurrency.propTypes = {
    sourceCurrency: PropTypes.string.isRequired,
    destinationCurrency: PropTypes.string.isRequired,
    inProgress: PropTypes.bool,
    wallet: PropTypes.object.isRequired,
    amount: PropTypes.number.isRequired,
    rate: PropTypes.number.isRequired,
    onChange: PropTypes.func.isRequired,
}

export default DestinationCurrency;
