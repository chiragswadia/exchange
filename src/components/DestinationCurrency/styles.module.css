.root {
    position: relative;
    margin-top: 16px;
  }
  
  .amount {
    position: absolute;
    top: 16px;
    right: 36px;
    height: 47px;
    max-width: 300px;
    overflow: visible;
    pointer-events: none;
  }
  
  .value {
    display: flex;
    align-items: flex-end;
    min-height: 47px;
    text-align: right;
    white-space: nowrap;
    font-size: 44px;
    color: #333;
  }
  
  .valueLong {
    font-size: 36px;
  }
  
  .value:not(:empty)::before {
    content: '+';
  }
  
  .rate {
    margin-top: 12px;
    text-align: right;
    font-size: 16px;
    color: #333;
  }
  
  
  