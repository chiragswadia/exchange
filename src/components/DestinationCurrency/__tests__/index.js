import React from "react";
import { cleanup } from '@testing-library/react';
import renderer from "react-test-renderer";
import DestinationCurrency from "../";

beforeEach(cleanup);

const wallet = {
    EUR: 100,
    USD: 200,
    GBP: 300,
}

describe("DestinationCurrency", () => {
    test("Snapshot should match #1", () => {
        const onChange = jest.fn();

        expect(
            renderer.create(
                <DestinationCurrency
                    wallet={wallet}
                    destinationCurrency="GBP"
                    onChange={onChange}
                />
            ).toJSON()
        ).toMatchSnapshot();
    });

    test("Snapshot should match #2", () => {
        const onChange = jest.fn();

        expect(
            renderer.create(
                <DestinationCurrency
                    wallet={wallet}
                    destinationCurrency="EUR"
                    onChange={onChange}
                />
            ).toJSON()
        ).toMatchSnapshot();
    });
});
