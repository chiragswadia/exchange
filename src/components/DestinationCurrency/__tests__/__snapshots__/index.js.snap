// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DestinationCurrency Snapshot should match #1 1`] = `
<div
  className="root"
>
  <div
    className="root"
  >
    <div>
      <div
        className="carousel carousel-slider"
        style={
          Object {
            "width": "100%",
          }
        }
      >
        <button
          aria-label="previous slide / item"
          className="control-arrow control-prev control-disabled"
          onClick={[Function]}
          type="button"
        />
        <div
          className="slider-wrapper axis-horizontal"
          style={Object {}}
        >
          <ul
            className="slider animated"
            onMouseDown={[Function]}
            onTouchEnd={[Function]}
            onTouchStart={[Function]}
            style={
              Object {
                "MozTransform": "translate3d(-300%,0,0)",
                "MozTransitionDuration": "350ms",
                "MsTransform": "translate3d(-300%,0,0)",
                "MsTransitionDuration": "350ms",
                "OTransform": "translate3d(-300%,0,0)",
                "OTransitionDuration": "350ms",
                "WebkitTransform": "translate3d(-300%,0,0)",
                "WebkitTransitionDuration": "350ms",
                "msTransform": "translate3d(-300%,0,0)",
                "msTransitionDuration": "350ms",
                "transform": "translate3d(-300%,0,0)",
                "transitionDuration": "350ms",
              }
            }
          >
            <li
              className="slide selected"
              onClick={[Function]}
            >
              <div
                className="root"
              >
                <div
                  className="currency"
                >
                  GBP
                </div>
                <div
                  className="walletBalance"
                  data-testid="wallet-balance"
                >
                  You have 
                   
                  <span
                    data-testid="formatted-currency"
                  >
                    £
                     
                    300.00
                  </span>
                </div>
              </div>
            </li>
            <li
              className="slide"
              onClick={[Function]}
            >
              <div
                className="root"
              >
                <div
                  className="currency"
                >
                  EUR
                </div>
                <div
                  className="walletBalance"
                  data-testid="wallet-balance"
                >
                  You have 
                   
                  <span
                    data-testid="formatted-currency"
                  >
                    €
                     
                    100.00
                  </span>
                </div>
              </div>
            </li>
            <li
              className="slide"
              onClick={[Function]}
            >
              <div
                className="root"
              >
                <div
                  className="currency"
                >
                  USD
                </div>
                <div
                  className="walletBalance"
                  data-testid="wallet-balance"
                >
                  You have 
                   
                  <span
                    data-testid="formatted-currency"
                  >
                    $
                     
                    200.00
                  </span>
                </div>
              </div>
            </li>
            <li
              className="slide selected"
              onClick={[Function]}
            >
              <div
                className="root"
              >
                <div
                  className="currency"
                >
                  GBP
                </div>
                <div
                  className="walletBalance"
                  data-testid="wallet-balance"
                >
                  You have 
                   
                  <span
                    data-testid="formatted-currency"
                  >
                    £
                     
                    300.00
                  </span>
                </div>
              </div>
            </li>
            <li
              className="slide"
              onClick={[Function]}
            >
              <div
                className="root"
              >
                <div
                  className="currency"
                >
                  EUR
                </div>
                <div
                  className="walletBalance"
                  data-testid="wallet-balance"
                >
                  You have 
                   
                  <span
                    data-testid="formatted-currency"
                  >
                    €
                     
                    100.00
                  </span>
                </div>
              </div>
            </li>
          </ul>
        </div>
        <button
          aria-label="next slide / item"
          className="control-arrow control-next control-disabled"
          onClick={[Function]}
          type="button"
        />
        <ul
          className="control-dots"
        >
          <li
            aria-label="slide item 1"
            className="dot"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
            value={0}
          />
          <li
            aria-label="slide item 2"
            className="dot"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
            value={1}
          />
          <li
            aria-label="slide item 3"
            className="dot selected"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
            value={2}
          />
        </ul>
      </div>
    </div>
  </div>
  <div
    className="amount"
  />
</div>
`;

exports[`DestinationCurrency Snapshot should match #2 1`] = `
<div
  className="root"
>
  <div
    className="root"
  >
    <div>
      <div
        className="carousel carousel-slider"
        style={
          Object {
            "width": "100%",
          }
        }
      >
        <button
          aria-label="previous slide / item"
          className="control-arrow control-prev control-disabled"
          onClick={[Function]}
          type="button"
        />
        <div
          className="slider-wrapper axis-horizontal"
          style={Object {}}
        >
          <ul
            className="slider animated"
            onMouseDown={[Function]}
            onTouchEnd={[Function]}
            onTouchStart={[Function]}
            style={
              Object {
                "MozTransform": "translate3d(-100%,0,0)",
                "MozTransitionDuration": "350ms",
                "MsTransform": "translate3d(-100%,0,0)",
                "MsTransitionDuration": "350ms",
                "OTransform": "translate3d(-100%,0,0)",
                "OTransitionDuration": "350ms",
                "WebkitTransform": "translate3d(-100%,0,0)",
                "WebkitTransitionDuration": "350ms",
                "msTransform": "translate3d(-100%,0,0)",
                "msTransitionDuration": "350ms",
                "transform": "translate3d(-100%,0,0)",
                "transitionDuration": "350ms",
              }
            }
          >
            <li
              className="slide"
              onClick={[Function]}
            >
              <div
                className="root"
              >
                <div
                  className="currency"
                >
                  GBP
                </div>
                <div
                  className="walletBalance"
                  data-testid="wallet-balance"
                >
                  You have 
                   
                  <span
                    data-testid="formatted-currency"
                  >
                    £
                     
                    300.00
                  </span>
                </div>
              </div>
            </li>
            <li
              className="slide selected"
              onClick={[Function]}
            >
              <div
                className="root"
              >
                <div
                  className="currency"
                >
                  EUR
                </div>
                <div
                  className="walletBalance"
                  data-testid="wallet-balance"
                >
                  You have 
                   
                  <span
                    data-testid="formatted-currency"
                  >
                    €
                     
                    100.00
                  </span>
                </div>
              </div>
            </li>
            <li
              className="slide"
              onClick={[Function]}
            >
              <div
                className="root"
              >
                <div
                  className="currency"
                >
                  USD
                </div>
                <div
                  className="walletBalance"
                  data-testid="wallet-balance"
                >
                  You have 
                   
                  <span
                    data-testid="formatted-currency"
                  >
                    $
                     
                    200.00
                  </span>
                </div>
              </div>
            </li>
            <li
              className="slide"
              onClick={[Function]}
            >
              <div
                className="root"
              >
                <div
                  className="currency"
                >
                  GBP
                </div>
                <div
                  className="walletBalance"
                  data-testid="wallet-balance"
                >
                  You have 
                   
                  <span
                    data-testid="formatted-currency"
                  >
                    £
                     
                    300.00
                  </span>
                </div>
              </div>
            </li>
            <li
              className="slide selected"
              onClick={[Function]}
            >
              <div
                className="root"
              >
                <div
                  className="currency"
                >
                  EUR
                </div>
                <div
                  className="walletBalance"
                  data-testid="wallet-balance"
                >
                  You have 
                   
                  <span
                    data-testid="formatted-currency"
                  >
                    €
                     
                    100.00
                  </span>
                </div>
              </div>
            </li>
          </ul>
        </div>
        <button
          aria-label="next slide / item"
          className="control-arrow control-next control-disabled"
          onClick={[Function]}
          type="button"
        />
        <ul
          className="control-dots"
        >
          <li
            aria-label="slide item 1"
            className="dot selected"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
            value={0}
          />
          <li
            aria-label="slide item 2"
            className="dot"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
            value={1}
          />
          <li
            aria-label="slide item 3"
            className="dot"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
            value={2}
          />
        </ul>
      </div>
    </div>
  </div>
  <div
    className="amount"
  />
</div>
`;
